### ISSUE TITLE
False positive in clang-tidy misc-const-correctness when returning pointer to non-const

### ISSUE THREAD

**Issue by JackH3103** [llvm/llvm-project#189190]
```cpp
#include <array>
#include <cstddef>
#include <iostream>

class my_class {
public:
    int *get_item(size_t i) {
        int *ret = nullptr;
        if (i < nums.size()) {
            ret = &nums[i];
        }
        return ret;
    }
private:
    std::array<int, 5> nums{1, 2, 3, 4, 5};
};

int main()
{
    my_class my_object;

    int *num = my_object.get_item(0);
    *num = 6;
    std::cout << *num << '\n';

    return 0;
}

```

Running `clang-tidy -checks='misc-const-correctness' -fix test.cpp` it tries to replace `int *ret = nullptr;` with `int  const*ret = nullptr;` which doesn't compile.

LLVM version 21.1.8
Operating System: openSUSE Tumbleweed 20260326
